godtools/Views/TractElements/Cards.swift

```diff
@@ -65,10 +65,32 @@ class Cards: BaseTractElement {
     
     override func setupView(properties: Dictionary<String, Any>) {
         self.frame = buildFrame()
+        setupBackground()
+    }
+    
+    func setupBackground() {
         self.backgroundColor = .clear
         
         let image = UIImage(named: self.backgroundImagePath)
         let imageView = UIImageView(image: image)
+        
+        let viewWidth = self.frame.size.width
+        let viewHeight = self.frame.size.height
+        var width = image?.size.width
```

Comment:
I would just unwrap image here. It will save having to unwrap the width and height everywhere you use it.

You could and should also do a guard let when loading the image to prevent crashes. So above you could do:
````swift
guard let image = UIImage(name: self.backgroundImagePath) else {
 return
}
````
This way the worst thing that can happen is there is no background if the image is missing. As it is now, the app will crash.